user.go

```diff
@@ -1058,6 +1065,12 @@ func (user *User) e2eeEventHandler(rawEvt any) {
 	}
 }
 
+func (user *User) resetWADeviceAndReconnect(reason events.ConnectFailureReason) {
+	user.log.Debug().Int("ConnectFailureReason", int(reason)).Msg("Resetting WADevice and reconnecting")
+	user.WADevice = nil
+	go user.FullReconnect()
```

Comment:
There isn't really any need for a full reconnect in the 415 case, just

```go
if user.E2EEClient != nil {
  user.E2EEClient.Disconnect()
  user.E2EEClient = nil
}
err := user.connectE2EE()
```

should be enough

(the 401 case probably does need a full reconnect though to check if the user got completely logged out or not)